Mathlib/MeasureTheory/Measure/LevyProkhorovMetric.lean

```diff
@@ -387,19 +435,220 @@ lemma continuous_levyProkhorov_to_probabilityMeasure :
+
+open BoundedContinuousFunction TopologicalSpace
+
+variable {ι : Type*} (Ω : Type*) [PseudoMetricSpace Ω] [SeparableSpace Ω]
+variable [MeasurableSpace Ω] [OpensMeasurableSpace Ω]
+
+/-- In a separable pseudometric space, for any ε > 0 there exists a countable collection of
+disjoint Borel measurable subsets of diameter at most ε that cover the whole space. -/
+lemma SeparableSpace.exists_measurable_partition_diam_le {ε : ℝ} (ε_pos : 0 < ε) :
+    ∃ (As : ℕ → Set Ω), (∀ n, MeasurableSet (As n)) ∧ (∀ n, Bornology.IsBounded (As n)) ∧
+        (∀ n, diam (As n) ≤ ε) ∧ (⋃ n, As n = univ) ∧
+        (Pairwise (fun (n m : ℕ) ↦ Disjoint (As n) (As m))) := by
+  by_cases X_emp : IsEmpty Ω
+  · refine ⟨fun _ ↦ ∅, fun _ ↦ MeasurableSet.empty, fun _ ↦ Bornology.isBounded_empty, ?_, ?_,
+            fun _ _ _ ↦ disjoint_of_subsingleton⟩
+    · intro n
+      simpa only [diam_empty] using LT.lt.le ε_pos
+    · simp only [iUnion_empty]
+      apply Eq.symm
+      simp only [univ_eq_empty_iff, X_emp]
+  rw [not_isEmpty_iff] at X_emp
+  obtain ⟨xs, xs_dense⟩ := exists_dense_seq Ω
+  have half_ε_pos : 0 < ε / 2 := half_pos ε_pos
+  set Bs := fun n ↦ Metric.ball (xs n) (ε/2)
+  set As := disjointed Bs
+  refine ⟨As, ?_, ?_, ?_, ?_, ?_⟩
+  · exact MeasurableSet.disjointed (fun n ↦ measurableSet_ball)
+  · exact fun n ↦ Bornology.IsBounded.subset isBounded_ball <| disjointed_subset Bs n
+  · intro n
+    apply (diam_mono (disjointed_subset Bs n) isBounded_ball).trans
+    convert diam_ball half_ε_pos.le
+    ring
+  · have aux : ⋃ n, Bs n = univ := by
+      convert DenseRange.iUnion_uniformity_ball xs_dense <| Metric.dist_mem_uniformity half_ε_pos
+      exact (ball_eq_ball' _ _).symm
+    simpa only [← aux] using iUnion_disjointed
+  · exact disjoint_disjointed Bs
+
+variable {Ω}
+
+lemma ProbabilityMeasure.toMeasure_add_pos_gt_mem_nhds (P : ProbabilityMeasure Ω)
+    {G : Set Ω} (G_open : IsOpen G) {ε : ℝ≥0∞} (ε_pos : 0 < ε) (ε_ne_top : ε ≠ ∞) :
+    ({Q | P.toMeasure G < Q.toMeasure G + ε} ∈ 𝓝 P) := by
+  by_cases easy : P.toMeasure G < ε
+  · exact eventually_of_forall (fun _ ↦ lt_of_lt_of_le easy le_add_self)
+  simp only [not_lt] at easy
+  have aux : P.toMeasure G - ε < liminf (fun Q ↦ Q.toMeasure G) (𝓝 P) := by
+    apply lt_of_lt_of_le (ENNReal.sub_lt_self (measure_lt_top _ _).ne _ _)
+        <| ProbabilityMeasure.le_liminf_measure_open_of_tendsto tendsto_id G_open
+    · exact (lt_of_lt_of_le ε_pos easy).ne.symm
+    · exact ε_pos.ne.symm
+  filter_upwards [gt_mem_sets_of_limsInf_gt (α := ℝ≥0∞) isBounded_ge_of_bot
+      (show P.toMeasure G - ε < limsInf ((𝓝 P).map (fun Q ↦ Q.toMeasure G)) from aux)] with Q hQ
+  simp only [preimage_setOf_eq, mem_setOf_eq] at hQ
+  convert ENNReal.add_lt_add_right ε_ne_top hQ
+  exact (tsub_add_cancel_of_le easy).symm
+
+lemma ProbabilityMeasure.continuous_toLevyProkhorov :
+    Continuous (ProbabilityMeasure.toLevyProkhorov (Ω := Ω)) := by
+
```

Comment:
No empty lines in proofs. The comments are already enough to separate the steps of the proof.